src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -807,7 +807,18 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
         blocksToDelete[0].header.number
       );
     }
-    await this.#database.batch(() => {
+    
+    // notify subscribers that logs have been rolled back
+    await Promise.all(
+      blocksToDelete.map(async block => {
+        const blockNum = block.header.number.toBuffer();
+
+        const logsToDelete = (await this.blockLogs.get(blockNum)).asRemoved();
+        await this.emit("blockLogs", logsToDelete);
```

Comment:
> I'm wondering if we should emit these in reverse order

I considered that, but figured that since the docs I could find don't specify any order, it's probably better for testing purposes if they're in forward order. Also it looks like per the geth link you shared that geth also emits them in forward order, unless there's some logic elsewhere that I missed that reverses them when the `removed` flag is set.